cmd/thanos/compact.go

```diff
@@ -54,6 +59,11 @@ func registerCompact(m map[string]setupFunc, app *kingpin.Application, name stri
 			*haltOnError,
 			*wait,
 			time.Duration(*retention),
+			map[int64]time.Duration{
```

Comment:
I feel like we could add proper Go enum for `ResLevels`, so

```
type ResLevel int64
const(
   ResLevel0 = ResLevel(0), 
   ...
)
```

We can than do `map[downsample.ResLevel]`. Any thoughts?